src/main/java/org/jabref/logic/ai/chatting/chathistory/ChatHistoryService.java

```diff
@@ -41,40 +45,65 @@
  *    JabRef, but for {@link BibEntry} it is definitely not).
  */
 public class ChatHistoryService implements AutoCloseable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ChatHistoryService.class);
+
     private final StateManager stateManager = Injector.instantiateModelOrService(StateManager.class);
 
     private final CitationKeyPatternPreferences citationKeyPatternPreferences;
 
     private final ChatHistoryStorage implementation;
 
-    private record ChatHistoryManagementRecord(Optional<BibDatabaseContext> bibDatabaseContext, ObservableList<ChatMessage> chatHistory) { }
+    private record ChatHistoryManagementRecord<T>(T object, Optional<BibDatabaseContext> bibDatabaseContext, ObservableList<ChatMessage> chatHistory) { }
 
-    private final Map<BibEntry, ChatHistoryManagementRecord> bibEntriesChatHistory = new HashMap<>();
-    private final Map<AbstractGroup, ChatHistoryManagementRecord> groupsChatHistory = new HashMap<>();
+    private final Map<String, ChatHistoryManagementRecord<BibEntry>> bibEntriesChatHistory = new HashMap<>();
```

Comment:
Note that you can also create a `TreeMap` with a customComparator (see https://stackoverflow.com/a/5453431/873282).

Maybe, the code could be cleaner.